src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyCaseBlocksInSwitchStatementAlone2.java

```diff
@@ -0,0 +1,119 @@
+/*
+RightCurly
+option = ALONE
+tokens = LITERAL_CASE
+
+*/
+package com.puppycrawl.tools.checkstyle.checks.blocks.rightcurly;
+
+public class InputRightCurlyCaseBlocksInSwitchStatementAlone2 {
+     public static void test9() {
+        int mode = 0;
+        switch (mode) {
+            case 0: int x = 1;
+            case 1: x = 1; break;
+            case 2: {
+                x =
+            1;} default : x = 5;         // violation '}' at column 15 should be alone on a line'
+        }
+    }
+
+    public static void test10() {
+        int mode = 0;
+        switch (mode) {
+            case 0: {
+
+            } case 1: int x = 1; break;    // violation '}' at column 13 should be alone on a line'
+            case 2: {
+
+            }
+            default : x = 5;
+        }
+    }
+     public static void test11() {
+        int mode = 0;
+        switch (mode) {
+            case 0: {
+
+            } case 1: int x = 1; break;   // violation '}' at column 13 should be alone on a line'
+            case 2: {
+
+            } default : x = 5;            // violation '}' at column 13 should be alone on a line'
+        }
+    }
+    public static void test12() {
+        int mode = 0;
+        switch (mode) {
+            case 0: {
+
+            int x = 1; } case 1: {int x = 1;} break;  // 2 violations
+        }
+    }
+    public static void test13() {
+        int mode = 0;
+        switch (mode) {
+            case 0: { int x = 1;
+
+            } case 1: { // violation '}' at column 13 should be alone on a line'
+
+            }
+            default : break;
+        }
+    }
+    public static void test14() {
+        int mode = 0;
+        switch (mode) {
+            case 0: {
+
+            } case 1: {  }         // 2 violations
+            default : {break;}
+        }
+    }
+
+    public static void test15() {
+        int mode = 0;
+        switch (mode) {case 0: { } case 1: {  } } // 2 violations
+    }
+
+    public static void test16() {
+        int mode = 0;
+        switch (mode) {
+            case 0: int x = 1; { } break;  // violation '}' at column 34 should be alone on a line'
```

Comment:
So are we violating anywhere in a case that has a `{}`? It doesn't have to start the case? Middle and End is violated too?

@romani @nrmancuso Please confirm if this is acceptable as the issue only shows `{}` that start the cases.

Reply:
@rnveach Yes now it is violating the rcurly anywhere should it be at the start only? this was the case before I didn't even think of this syntax I was working on as we either put one block with braces `case x: {.....} `
or don't put braces at all `case x:.. `
but after your comment above I reconsidered that we can have the braces in the middle of the block
Please tell me what is the appropriate action here should I just leave this example but no violation?

Reply:
I leave this to the issue approvers I pinged. This seems like a weird case where we pick up all braces inside a case. For a similar train of thought, does this mean if we configure this check for "METHOD_DEF", then all solo braces inside a method itself should also be handled by this token. It is not an accurate analogy since they would be braces inside braces, but also technically the braces in the case tokens are not even connected to the case as shown in the AST. They are just a pure `SLIST`.

@mahfouz72 Can you add other test cases to show how your changes will fair against braces inside braces. Example: `case X: { { int x = 0; } }`. Are the inners validated as well?

Reply:
When I see examples using braces on `->` for cases, I agree that we should only be doing it when the braces surround the entire case (including any break). I am assuming it is invalid syntax to have anything after the braces in this case like `case X -> { int x = 0; } break;`. I don't use this new syntax, so I am less familiar with it.

I still await confirmation from others on non `->` but I am leaning towards having it be handled by `->` even if `->` can't be written like that.

Reply:
@mahfouz72 LeftCurly ( https://checkstyle.org/checks/blocks/leftcurly.html#LeftCurly ) supports `case`s. Please check if there is any conflicts on how you handle `}` versus how it handles `{`. If both line up, I am more inclined to approve the implementation.

Reply:
These blocks (`{}`) should not be checked. You can pretty much put blocks anywhere in Java.

Reply:
> Please check if there is any conflicts on how you handle `}` versus how it handles `{`

https://github.com/checkstyle/checkstyle/blob/f4b71587d76f15661e97c571a6183cb28d73b9eb/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/LeftCurlyCheck.java#L301-L314

the left curly is checking SLIST if it is the first child only



> `case X: { { int x = 0; } }`. Are the inners validated as well

no,  the inners are not validated. the `SLIST` that is a direct child of the main `SLIST` of the case only is validated but this is a nested SLIST (grandchild) and not checked



> but I am leaning towards having it be handled like `->`

I agree. to be in sync with leftcurly and with `->` I see it is better to check the braces only when it is the first child of the case any other inner braces should be ignored as it is not directly connected to the case

Reply:
> the left curly is checking SLIST if it is the first child only

Then I think this is how this check's implementation should be.

This issue started because Left check supports more tokens than Right check. If they are going to support the same tokens, they should validate the same cases. It doesn't make sense for one to support more or less than the other. It will have to be a new issue if they support case that we feel should not be supported, and vice versa.

Reply:
> Then I think this is how this check's implementation should be

yes I pushed update to validate onlynthe rculy that is a first child in case

Reply:
> the left curly is checking SLIST if it is the first child only

Just confirming, so as long as SLIST is the first child we will validate it even if there is more which comes after the `}`? It could be more code, break, or just the next case/default, etc...?

Reply:
@rnveach Yes exactly